Add an AI disclosure policy (#450)
dwavesystems/dwave-ocean-sdk: docs/ocean/ai_policy.rst

Responsibility
--------------

You are responsible for any code you submit to D-Wave's repositories, regardless
of whether it was manually written or generated by AI. You must understand and be able
to explain the code you submit as well as the existing related code. It is not
acceptable to submit a patch that you cannot understand and explain yourself.
In explaining your contribution, do not use AI to automatically generate
descriptions.

Disclosure
----------

You must disclose whether AI has been used to assist in the development of
your pull request.
If so, you must document which tool(s) have been used, how they were used,
and specify what code or text is AI generated. We will reject any pull request
that does not include the disclosure.

Code Quality
------------

Code generated by AI can be of low quality. Contributors are expected to
submit code that meets D-Wave's standards. We will reject pull requests that we
deem being "`AI slop`_". Do not waste developers' time by submitting code that
is fully or mostly generated by AI, and doesn't meet our standards.

.. _AI slop: https://en.wikipedia.org/wiki/AI_slop

Copyright
---------

> All open source code in Ocean is released under the Apache-2.0 copyright license.

Reviewer: ```suggestion
All open-source code in the Ocean SDK is released under the Apache-2.0 copyright license.
```

To keep Ocean as an adjective

> Contributors to Ocean license their code under the same license when it is

Reviewer: ```suggestion
Contributors to Ocean software license their code under the same license when it is
```

> included into Ocean's version control repository. That means contributors must

Reviewer: ```suggestion
included into the Ocean SDK's version control repository. That means contributors must
```

> own the copyright of any code submitted to Ocean or must include the Apache-2.0

Reviewer: ```suggestion
own the copyright of any code submitted to the Ocean SDK or must include the Apache-2.0
```

> own the copyright of any code submitted to Ocean or must include the Apache-2.0

Reviewer: "own the copyright"

Any concern that some devs might think this requires action on their part beyond independently developing the code, and that we should spell that out?

Author: Given that this section gets somewhere adjacent to the area adjacent to legalese, I am inclined to keep it as short as possible lest we accidentally contradict ourselves through trying to over-explain.

compatible open source license(s) associated with the submitted code
in the patch. Code generated by AI may infringe on copyright and it is the
submitter's responsibility to not infringe. We reserve the right to reject any pull
requests, AI generated or not, where the copyright is in question.

Communication
-------------

When interacting with developers (forum, discussions,
issues, pull requests, etc.) do not use AI to speak for you, except for
translation or grammar editing. If the developers want to chat with a chatbot,
they can do so themselves. Human-to-human communication is essential for an

> open source community to thrive.

Reviewer: ```suggestion
open-source community to thrive.
```
